src/modules/dynamic-Todo.js

```diff
@@ -0,0 +1,188 @@
+      py-1
+      bg-slate-100
+      text-gray-400
+      font-medium
+      text-xs
+      leading-tight
+      rounded
+      shadow-md
+      hover:bg-slate-300 hover:shadow-lg
+      focus:bg-slate-300 focus:shadow-lg focus:outline-none focus:ring-0
+      active:bg-slate-300 active:shadow-lg active:text-white
+      transition
+      duration-150
+      ease-in-out
+      flex
+      items-center
+      whitespace-nowrap`;
+
+        const dropItemDel = document.createElement('li');
+        const delBtn = document.createElement('button');
+        delBtn.type = 'button';
+        delBtn.innerHTML = `<svg xmlns="http://www.w3.org/2000/svg" class="h-5 w-5" fill="none" viewBox="0 0 24 24" stroke="currentColor" stroke-width="2">
+      <path stroke-linecap="round" stroke-linejoin="round" d="M15 12H9m12 0a9 9 0 11-18 0 9 9 0 0118 0z" />
+    </svg> 
+      <span>DEL</span>`;
+        delBtn.classList = `del-btn dropdown-item flex items-center space-x-2
+      text-sm
+      py-2
+      px-4
+      font-normal
+      w-full
+      whitespace-nowrap
+      bg-transparent
+      text-gray-700
+      transition
+      duration-150
+      hover:bg-red-500 hover:text-white`;
+        delBtn.onclick = () => todoInstance.onDelete(id);
+        dropItemDel.appendChild(delBtn);
+
+        const dropItemEdit = document.createElement('li');
+        const editBtn = document.createElement('button');
+        editBtn.type = 'button';
+        editBtn.innerHTML = `<svg xmlns="http://www.w3.org/2000/svg" class="h-5 w-5" fill="none" viewBox="0 0 24 24" stroke="currentColor" stroke-width="2">
+      <path stroke-linecap="round" stroke-linejoin="round" d="M11 5H6a2 2 0 00-2 2v11a2 2 0 002 2h11a2 2 0 002-2v-5m-1.414-9.414a2 2 0 112.828 2.828L11.828 15H9v-2.828l8.586-8.586z" />
+    </svg>
+    <span>EDIT</span>`;
+        editBtn.classList = `edit-btn dropdown-item flex items-center space-x-2
+      text-sm
+      py-2
+      px-4
+      font-normal
+      w-full
+      whitespace-nowrap
+      bg-transparent
+      text-gray-700
+      transition
+      duration-150
+      hover:bg-amber-500 hover:text-white`;
+        editBtn.onclick = () => todoInstance.onEdit(id);
+        dropItemEdit.appendChild(editBtn);
+
+        const dropmenu = document.createElement('ul');
+        dropmenu.setAttribute('aria-labelledby', `dropBtn ${id}`);
+        dropmenu.classList = `dropdown-menu
+      min-w-max
+      absolute
+      hidden
+      bg-white
+      text-base
+      z-50
+      float-left
+      right-full
+      py-2
+      list-none
+      text-left
+      rounded-lg
+      shadow-lg
+      mt-1
+      m-0
+      bg-clip-padding
+      border-none`;
+        dropmenu.append(dropItemDel, dropItemEdit);
+
+        const dropdown = document.createElement('div');
+        dropdown.classList = 'dropstart relative';
+        dropdown.append(dotBtn, dropmenu);
+
+        todoContainer.append(todoLabel, dropdown);
+      }
+      todoList.appendChild(todoContainer);
```

Comment:
The render function stretches more than 50 lines of code. Consider breaking the function into smaller functions to ease of reading.

Consider the following functions;
- Function to create "delBtn"
 - Function to create "editBtn"
 - Function to create "dotBtn"

Reply:
Noted I will look into it, thank you for the feedback 😄